src/sas/qtgui/Plotting/Slicers/BoxSlicer.py

```diff
@@ -19,50 +19,78 @@ class BoxInteractor(BaseInteractor, SlicerModel):
     function of Q_x and BoxInteractorY averages all the points from
     -x to +x as a function of Q_y
     """
-
-    def __init__(self, base, axes, item=None, color='black', zorder=3):
+    def __init__(self, base, axes, item=None, color='black', zorder=3, direction=None):
         BaseInteractor.__init__(self, base, axes, color=color)
         SlicerModel.__init__(self)
         # Class initialization
         self.markers = []
         self.axes = axes
         self._item = item
-        # connecting artist
+        #connecting artist
         self.connect = self.base.connect
         # which direction is the preferred interaction direction
-        self.direction = None
+        self.direction = direction
         # determine x y  values
-        self.x = 0.5 * min(numpy.fabs(self.data.xmax),
-                           numpy.fabs(self.data.xmin))
-        self.y = 0.5 * min(numpy.fabs(self.data.xmax),
-                           numpy.fabs(self.data.xmin))
-        # when reach qmax reset the graph
-        self.qmax = max(self.data.xmax, self.data.xmin,
-                        self.data.ymax, self.data.ymin)
+        if self.direction == "Y":
+            self.xwidth = 0.1 * (self.data.xmax - self.data.xmin) / 2
+            self.ywidth = 1.0 * (self.data.ymax - self.data.ymin) / 2
+            # when reach qmax reset the graph
+            self.qmax = max(numpy.fabs(self.data.ymax), numpy.fabs(self.data.ymin))
+        else:
```

Comment:
This case holds where `self.direction != "Y"`. Is this what we really want? Does case (y vs. Y) matter? What should happen if something other than X or Y is passed?